SYCL/ESIMD/api/functional/functions/functions_select_2d.hpp

```diff
@@ -0,0 +1,286 @@
+
+#pragma once
+
+#include "../shared_element.hpp"
+#include "common.hpp"
+
+namespace esimd = sycl::ext::intel::esimd;
+
+namespace esimd_test::api::functional::functions {
+
+// Descriptor class for the case of calling simd<T,N>::select function.
+struct select_2d {
+  static std::string get_description() { return "2d select"; }
+
+  template <typename DataT, int NumElems, int SizeY, int StrideY, int SizeX,
+            int StrideX, int Height, int Width>
+  static void call_esimd_function(const DataT *const ref_1,
+                                  const DataT *const ref_2, DataT *const out,
+                                  size_t offset_x, size_t offset_y) {
+    esimd::simd<DataT, NumElems> src_simd_obj;
+    src_simd_obj.copy_from(ref_1);
+
+    auto simd_view_instance =
+        src_simd_obj.template bit_cast_view<DataT, Height, Width>();
+    auto selected_elems =
+        simd_view_instance.template select<SizeY, StrideY, SizeX, StrideX>(
+            offset_y, offset_x);
+
+    size_t ref_2_index = 0;
+    for (int i = 0; i < SizeY; ++i) {
+      for (int j = 0; j < SizeX; ++j) {
+        auto a = selected_elems.row(i);
+        a[j] = ref_2[ref_2_index];
+        ++ref_2_index;
+      }
+    }
+
+    src_simd_obj.copy_to(out);
+  }
+};
+
+// The main test routine.
+// Using functor class to be able to iterate over the pre-defined data types.
+template <typename TestCaseT, typename SizeXT, typename SizeYT,
+          typename StrideXT, typename StrideYT, typename OffsetXT,
+          typename OffsetYT, typename SimdSplitCoeffT, typename DataT,
+          typename DimT>
+class run_test {
+  static constexpr int NumElems = DimT::value;
+  static constexpr int SizeY = SizeYT::value;
+  static constexpr int StrideY = StrideYT::value;
+  static constexpr int SizeX = SizeXT::value;
+  static constexpr int StrideX = StrideXT::value;
+  static constexpr int OffsetX = OffsetXT::value;
+  static constexpr int OffsetY = OffsetYT::value;
+  static constexpr int Height = NumElems / SimdSplitCoeffT::value;
+  static constexpr int Width = NumElems / Height;
+
+public:
+  bool operator()(sycl::queue &queue, const std::string &data_type) {
+    bool passed = true;
+
+    static_assert(NumElems == Height * Width,
+                  "Invalid SimdSplitCoeff, height and width multiplacation "
+                  "should be equal to simd size.");
+    static_assert(Width > SizeX, "Number selected elements can't be greater "
+                                 "than number elements in one line");
+    static_assert(
+        Height > SizeY,
+        "Number selected elements can't be greater than lines number");
+    assert(Height > OffsetY &&
+           "Offset of height should be less than simd_view height size");
+    assert(Width > OffsetX &&
+           "Offset of with should be less than simd_view with size");
+
+    constexpr size_t value_for_increase_ref_data_for_fill = 50;
+    static_assert(std::numeric_limits<char>::max() >
```

Comment:
One more https://github.com/intel/llvm-test-suite/pull/875#discussion_r822371818

Reply:
Naming and other cases has been fixed in this commit: https://github.com/intel/llvm-test-suite/pull/908/commits/ce54fc2c173d7fdf6944b2e91334eba4dbf7f90c